hammer/identification/lambdas/ecs-external-image-source-issues-identification/describe_ecs_external_image_source_issues.py

```diff
@@ -0,0 +1,88 @@
+    try:
+        payload = json.loads(event["Records"][0]["Sns"]["Message"])
+        account_id = payload['account_id']
+        account_name = payload['account_name']
+        # get the last region from the list to process
+        region = payload['regions'].pop()
+        # region = payload['region']
+    except Exception:
+        logging.exception(f"Failed to parse event\n{event}")
+        return
+
+    try:
+        config = Config()
+
+        main_account = Account(region=config.aws.region)
+        ddb_table = main_account.resource("dynamodb").Table(config.ecs_external_image_source.ddb_table_name)
+
+        account = Account(id=account_id,
+                          name=account_name,
+                          region=region,
+                          role_name=config.aws.role_name_identification)
+        if account.session is None:
+            return
+
+        logging.debug(f"Checking Image source is external or internal for ecs task definitions in {account}")
+
+        # existing open issues for account to check if resolved
+        open_issues = IssueOperations.get_account_open_issues(ddb_table, account_id, ECSExternalImageSourceIssue)
+        # make dictionary for fast search by id
+        # and filter by current region
+        open_issues = {issue.issue_id: issue for issue in open_issues if issue.issue_details.region == region}
+        logging.debug(f"ECS task definitions in DDB:\n{open_issues.keys()}")
+
+        checker = ECSChecker(account=account)
+        if checker.check():
+            for task_definition in checker.task_definitions:
+                logging.debug(f"Checking {task_definition.name}")
+                if task_definition.external_image:
+                    issue = ECSExternalImageSourceIssue(account_id, task_definition.name)
+                    issue.issue_details.arn = task_definition.arn
+                    issue.issue_details.tags = task_definition.tags
+                    issue.issue_details.container_name = task_definition.container_name
+                    issue.issue_details.image_url = task_definition.image_url
+                    issue.issue_details.region = task_definition.account.region
+                    if config.ecs_external_image_source.in_whitelist(account_id, task_definition.name):
+                        issue.status = IssueStatus.Whitelisted
+                    else:
+                        issue.status = IssueStatus.Open
+                    logging.debug(f"Setting {task_definition.name} status {issue.status}")
+                    IssueOperations.update(ddb_table, issue)
+                    # remove issue id from issues_list_from_db (if exists)
+                    # as we already checked it
+                    open_issues.pop(task_definition.name, None)
+
+        logging.debug(f"ECS task definitions in DDB:\n{open_issues.keys()}")
+        # all other unresolved issues in DDB are for removed/remediated task definitions
+        for issue in open_issues.values():
```

Comment:
I think we should track progress of the scan triggered from our API, example: https://github.com/dowjones/hammer/blob/master/hammer/identification/lambdas/ami-public-access-issues-identification/describe_public_ami_issues.py#L76